src/main/java/seedu/address/logic/commands/EditTypeCommand.java

```diff
@@ -48,24 +47,13 @@ protected static Schedule createEditedSchedule(
         assert scheduleToEdit != null;
 
         List<Event> scheduleEvents = scheduleToEdit.getEvents();
-        ArrayList<Event> updatedEvents = new ArrayList<>();
-        EditScheduleDescriptor editScheduleDescriptor = new EditScheduleDescriptor();
-
-        int counter = 0;
-        for (Event toEditEvent : scheduleEvents) {
-            Event updatedEvent;
-            if (counter == targetEventIndex.getZeroBased()) {
-                updatedEvent = createEditedEvent(toEditEvent, editEventDescriptor);
-            } else {
-                updatedEvent = toEditEvent;
-            }
-            updatedEvents.add(updatedEvent);
-            counter++;
-        }
-
-        editScheduleDescriptor.setEvents(updatedEvents);
-
-        return new Schedule(editScheduleDescriptor.getEvents());
+        ArrayList<Event> updatedEvents = new ArrayList<>(scheduleEvents);
+
+        Event toEditEvent = updatedEvents.remove(targetEventIndex.getZeroBased());
+        Event updatedEvent = createEditedEvent(toEditEvent, editEventDescriptor);
+        updatedEvents.add(targetEventIndex.getZeroBased(), updatedEvent);
```

Comment:
Remove and re-adding the event moves it to the bottom of the index right? Would it be more user friendly to add it back to its original index using add(index, eventToAdd) instead to keep the indexes the same?